cmake/modules/RootConfiguration.cmake

```diff
@@ -599,6 +599,14 @@ else()
   set(hasstdinvoke undef)
 endif()
 
+CHECK_CXX_SOURCE_COMPILES("#include <utility>
+int main() { std::make_index_sequence<3> s; return 0; }" found_stdindexsequence)
```

Comment:
Should we test getting the value (i.e. making minimally sure the 'api' is the one expected?

Reply:
Are you saying there are standard library implementations with a `std::make_index_sequence` implementation that _does not_ return the expected `std::index_sequence` and that we want to support them anyway?

At that point nothing is real anymore :sweat_smile:

I can make it
```c++
CHECK_CXX_SOURCE_COMPILES("#include <utility>
#include <type_traits>
int main() { static_assert(std::is_same<std::integer_sequence<std::size_t, 0, 1, 1>, std::make_index_sequence<3>>::value, ""); return 0; }"
```
would that be ok?

Reply:
Looks fine.  Probably even more than I asked :).

Reply:
I wasn't sure what you meant with "test getting the value".
I added the check.